docs/articles/documentation/how-it-works/README.md

```diff
@@ -3,151 +3,40 @@ layout: docs
 
-### Client-server architecture
+Our experience shows that this hybrid approach gives users the best of both worlds. Early versions of TestCafe ran entirely in the browser. This caused a number of issues. The tests’ capabilities were limited, the application often interfered with page execution, and the sub-optimal logic separation caused browser crashes. Since then, we decoupled the engine, exposed the underlying Node.js API, and implemented a test runner that executes test scripts server-side.
 
-TestCafe runs test code written by the user in the Node.js environment on the "server side". Scripts that emulate user activity (e.g., *automation scripts*) run inside the tested page in the browser on the same or on a different device. This is the "client side".
+## Page Proxying: Bridging the Server-Client Gap
 
-With this hybrid approach, TestCafe tests combine advantages of the browser scripts (built-in waiting, mobile testing) and standalone server-side code (access to the app's server and testing environment, better control over test execution).
+You may notice that when you run TestCafe, the URL in the browser’s address bar does not match that of your website. This happens because TestCafe runs an under-the-hood reverse proxy.
 
-### Node.js API
+The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and adds automation scripts to the requested resources. It then modifies the URLs contained within the resource so that they point to the proxy. That way, neither the client-side code nor the resources it communicates with can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe mocks the browser API.
 
-Test code uses compact API that consists of just a few dozen methods that eventually transform to user actions on the page. Since this server-side code runs in Node.js, tests should be written in JavaScript or a language TestCafe can compile to JavaScript (like TypeScript or CoffeeScript).
+The proxying mechanism ensures that the page appears to be hosted at the original URL even to the test code. This is why you can use your website’s actual URL in tests, and pay no mind to the browser’s address bar.
 
-The concise API makes tests easy to write, read, and maintain. Support for popular web languages allows most users to use their existing skills and minimizes the learning curve for new users.
+## Browser Sandboxing
 
-### Page proxying
+Every TestCafe test begins with an empty slate of a browser. At the end of a run, TestCafe deletes all browser cookies, empties the storage, and reloads the page, thereby preventing undesirable interference with subsequent tests. You don’t need to write any boilerplate code to reset the app state and reverse the changes your tests make.
```

Comment:
```an empty slate of a browser```

What is `slate`? My dictionary does not give any good translations for this word. Did you mean `state`?